backend/src/test/java/de/neuefische/backend/controller/AnimalControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package de.neuefische.backend.controller;
+
+import de.neuefische.backend.dto.AnimalDTO;
+import de.neuefische.backend.models.AnimalData;
+import de.neuefische.backend.services.AnimalService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.client.HttpServerErrorException;
+
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class AnimalControllerTest {
```

Comment:
Why do you use mockMvc instead of @SpringBootTest? 😄

Reply:
You can if you want and know the difference... Maybe you can do a test with mockMvc and one full SpringBootTest 👍